imagickkernel_class.c

```diff
@@ -641,10 +641,11 @@ PHP_METHOD(imagickkernel, addkernel)
 		RETURN_NULL();
 	}
 
-	do {
+	kernel_info = internp->kernel_info;
+	while (kernel_info != NULL) {
 		kernel_info_target = kernel_info;
 		kernel_info = kernel_info->next;
-	} while (kernel_info != NULL);
+	};
 
 	kernel_info_add_clone = CloneKernelInfo(kernel->kernel_info);
 	kernel_info_target->next = kernel_info_add_clone;
```

Comment:
What if while {} is never entered because kernel_info==NULL? kernel_info_target will be uninitialized! That case must be handled - best before executing the do/while.

Reply:
Thanks, you are right. Reverted back to do/while loop and added only change line.